app/models/ZkKafka.scala

```diff
@@ -60,9 +60,10 @@ object ZkKafka {
         val path = makePath(applyBase(Seq(stormZkRoot, Some(root))) ++ Seq(Some(spout)) ++ Seq(Some(part)))
         // Use the first partition's data to build up info about the topology
         // Also, don't trust zk to have valid JSON, it may be null or something...
-        getZkData(path).map { zkData =>
-          val jsonState = new String(zkData)
-          val state = Json.parse(jsonState)
+        for {
```

Comment:
Question: Out of curiosity, do you have any high-level guidance on when to prefer `for` comprehensions vs. `map`?